File: scalaz/src/main/scala/org/openapitools/client/api/ResponseGeocodingProperties.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import ResponseGeocodingProperties._

case class ResponseGeocodingProperties (
  name: String,
label: String,
score: Option[Double],
houseNumber: Option[String],
street: Option[String],
region: Option[String],
regionCode: Option[String],
neighbourhood: Option[String],
county: Option[String],
macroregion: Option[String],
city: Option[String],
country: Option[String],
countryCode: Option[String],
continent: Option[String],
postcode: Option[String],
features: Option[ResponseMapInfoFeatures])

object ResponseGeocodingProperties {
  import DateTimeCodecs._

  implicit val ResponseGeocodingPropertiesCodecJson: CodecJson[ResponseGeocodingProperties] = CodecJson.derive[ResponseGeocodingProperties]
  implicit val ResponseGeocodingPropertiesDecoder: EntityDecoder[ResponseGeocodingProperties] = jsonOf[ResponseGeocodingProperties]
  implicit val ResponseGeocodingPropertiesEncoder: EntityEncoder[ResponseGeocodingProperties] = jsonEncoderOf[ResponseGeocodingProperties]
}

File: clojure/src/travel_time_platform_api/specs/response_geocoding_properties.clj
(ns travel-time-platform-api.specs.response-geocoding-properties
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [travel-time-platform-api.specs.response-map-info-features :refer :all]
            )
  (:import (java.io File)))


(def response-geocoding-properties-data
  {
   (ds/req :name) string?
   (ds/req :label) string?
   (ds/opt :score) float?
   (ds/opt :house_number) string?
   (ds/opt :street) string?
   (ds/opt :region) string?
   (ds/opt :region_code) string?
   (ds/opt :neighbourhood) string?
   (ds/opt :county) string?
   (ds/opt :macroregion) string?
   (ds/opt :city) string?
   (ds/opt :country) string?
   (ds/opt :country_code) string?
   (ds/opt :continent) string?
   (ds/opt :postcode) string?
   (ds/opt :features) response-map-info-features-spec
   })

(def response-geocoding-properties-spec
  (ds/spec
    {:name ::response-geocoding-properties
     :spec response-geocoding-properties-data}))
